src/System.Private.CoreLib/src/System/Resources/ResourceManager.cs

```diff
@@ -225,9 +225,6 @@ public ResourceManager(string baseName, Assembly assembly)
             if (null == assembly)
                 throw new ArgumentNullException(nameof(assembly));
 
-            if (!(assembly is RuntimeAssembly))
```

Comment:
What happens when somebody initializes the ResourceManager with non-runtime Assembly now that we do not have this check?

Reply:
The only other concrete implementation is `AssemblyBuilder`. In its case it will just forward the calls to the `InternalAssembly`, which will throw appropriate exceptions. In general it will call the public APIs on the `Assembly` class now.

Reply:
The only missing thing I didn't implement for non-RuntimeAssembly types is the `GetNeutralResourcesLanguageAttribute` method. I can easily do that though since the implementation already exists in CoreRT.

Reply:
Could you please create PR with a test in CoreFX that is exercising the generic codepath? The PR should  fail without this change and it should pass with this change.

Reply:
I'll try to make something up. CoreRT already behaved that way, so I guess no additional tests there, or should I make tests for both?

Reply:
I assume the tests will go to `src/System.Resources.ResourceManager/tests` in CoreFX, right? Btw, I checked that the condition for `RuntimeAssembly` was added somewhere in .NET 4.x timeframe, but it was not present in .NET 2.0 (and all versions up to 3.5 consequently).

Reply:
> `src/System.Resources.ResourceManager/tests` in CoreFX, right? 

Right.

> condition for RuntimeAssembly was added somewhere in .NET 4.x 

The reflection object model was non-extensible in .NET 2.x. The only instance of Assembly that could ever exist was the RuntimeAssembly at that time.

Reply:
> The reflection object model was non-extensible in .NET 2.x. The only instance of Assembly that could ever exist was the RuntimeAssembly at that time.

Ah, I see. I missed that.

Reply:
Opened PR with couple of very simple tests. They pass on my CoreCLR builds locally.